app/src/main/java/ch/sdp/vibester/database/DataGetter.kt

```diff
@@ -210,14 +214,19 @@ class DataGetter @Inject constructor() {
 
     /**
      * This functions fetches the data of the given user from the database
-     * @param roomName the name of the room to retrieve data from
-     * @param callback the function to be called when the data of the appropriate user is available
+     * @param roomID the name of the room to retrieve data from
+     * @param partyRoomCallback the function to be called when the data of the appropriate room data is available
+     * @param songListCallback the function to be called when the data of the appropriate song list is available
+
      */
 
-    fun getRoomData(roomName: String, callback: (PartyRoom) -> Unit) {
+    fun getRoomData(roomID: String,
+                    partyRoomCallback: (PartyRoom, String) -> Unit,
+                    songListCallback: (MutableList<Pair<String, String>>) -> Unit) {
+
         val queryRooms = dbRoomRef
-            .orderByChild("roomName")
-            .equalTo(roomName)
+            .orderByChild("roomID")
+            .equalTo(roomID)
```

Comment:
It's good that you changed the name of the extra from "roomName" to "roomId", it makes it clear that we're going to use this value to uniquely identify rooms.